### PR TITLE
adding a test to verify that nodes belonging to different sporks cann…

### PR DESCRIPTION
**Pull request by vishalchangrani**
…ot talk to each other using 1-k messaging

This PR adds a test to verify that if there are two nodes belonging to different sporks, then they cannot communicate with each other using 1-k messaging. A similar test for 1-1 messaging already exists.

### CODE REVIEW

File: network/gossip/libp2p/sporking_test.go [hunk 1]
```diff
@@ -118,18 +107,126 @@ func (h *SporkingTestSuite) TestProtocolChangeAfterHardSpoon() {
...
+	err = node1.UnSubscribe(topicBeforeSpork)
+	assert.NoError(h.T(), err)
+	_, err = node1.Subscribe(ctx, topicAfterSpork)
+	assert.NoError(h.T(), err)
+
+	// assert that node 1 can no longer send a message to node 2 via PubSub
+	testOneToKMessagingIsNotAllowed(ctx, h.T(), node1, sub2, topicAfterSpork)
+}
+
+func testOneToOneMessagingIsAllowed(t *testing.T, sourceNode *P2PNode, dstnAddress NodeAddress) {
```
Comment (vishalchangrani) [inline network/gossip/libp2p/sporking_test.go:181]: commonizing the checks used by all 3 tests here

File: network/gossip/libp2p/sporking_test.go [hunk 2]
```diff
@@ -118,18 +107,127 @@ func (h *SporkingTestSuite) TestProtocolChangeAfterHardSpoon() {
...
+	payload := []byte("hello")
+	err := sourceNode.Publish(ctx, topic, payload)
+	assert.NoError(t, err)
+
+	// assert that the message is received by the destination node
+	unittest.AssertReturnsBefore(t, func() {
+		msg, err := dstnSub.Next(ctx)
+		assert.NoError(t, err)
+		assert.Equal(t, payload, msg.Data)
+	}, 5*time.Second)
```
Comment (jordanschalm) [inline network/gossip/libp2p/sporking_test.go:215]: We have 5 seconds here to check a successful message but only 1 second to check a failed message. If it takes say 2 seconds to send a 1-k message, `testOneToKMessagingFails` will pass even if the message does go through. 

Do you know how long the message publish takes in practice?

File: network/gossip/libp2p/sporking_test.go [hunk 3]
```diff
@@ -118,18 +107,127 @@ func (h *SporkingTestSuite) TestProtocolChangeAfterHardSpoon() {
...
+
+	// create and start node 1 on localhost and random port
+	node1, _ := h.CreateNode("node1", node1key, "0.0.0.0", "0", rootIDBeforeSpork, nil, false)
+	defer h.StopNode(node1)
+
+	// create and start node 2 on localhost and random port with the same root block ID
+	node2key, err := generateNetworkingKey("def")
+	assert.NoError(h.T(), err)
+	node2, addr2 := h.CreateNode("node1", node2key, "0.0.0.0", "0", rootIDBeforeSpork, nil, false)
+	defer h.StopNode(node1)
```
Comment (jordanschalm) [inline network/gossip/libp2p/sporking_test.go:140]:
```suggestion
	defer h.StopNode(node2)
```

File: network/gossip/libp2p/sporking_test.go (hunk 2 above)
Comment (vishalchangrani) [reply network/gossip/libp2p/sporking_test.go:215]: In practice is should take way under a second once the mesh is formed. I agree, the timeout for the failed message should also be large (atleast 1.5 second). 
Lemme fine tune those numbers a little

**Review by jordanschalm — APPROVED** [review]

File: network/gossip/libp2p/sporking_test.go [hunk 4]
```diff
@@ -118,18 +108,129 @@ func (h *SporkingTestSuite) TestProtocolChangeAfterHardSpoon() {
...
+
+	// create and start node 1 on localhost and random port
+	node1key, err := generateNetworkingKey("abc")
+	assert.NoError(h.T(), err)
+	node1, _ := h.CreateNode("node1", node1key, "0.0.0.0", "0", rootIDBeforeSpork, nil, false)
+	defer h.StopNode(node1)
+
+	// create and start node 2 on localhost and random port with the same root block ID
+	node2key, err := generateNetworkingKey("def")
+	assert.NoError(h.T(), err)
```
Comment (yhassanzadeh13) [inline network/gossip/libp2p/sporking_test.go:136]: suggest `require.NoError` instead, also other places in the code.

**Review by yhassanzadeh13 — APPROVED** [review]

File: network/gossip/libp2p/sporking_test.go (hunk 4 above)
Comment (vishalchangrani) [reply network/gossip/libp2p/sporking_test.go]: changed all assert.NoError to require.NoError

**Pull request merged**